lib/matplotlib/collections.py

```diff
@@ -106,9 +91,67 @@ def __init__(self,
                  **kwargs
                  ):
         """
-        Create a Collection
+        Parameters
+        ----------
+        edgecolors : color or list of colors, default: :rc:`patch.edgecolor`
+            Edge color for each patch making up the collection. The special
+            value 'face' can be passed to make the edgecolor match the
+            facecolor.
+        facecolors : color or list of colors, default: :rc:`patch.facecolor`
+            Face color for each patch making up the collection.
+        linewidths : float or list of floats, default: :rc:`patch.linewidth`
+            Line width for each patch making up the collection.
+        linestyles : str or tuple or list thereof, default: 'solid'
+            Valid strings are ['solid', 'dashed', 'dashdot', 'dotted', '-',
+            '--', '-.', ':']. Dash tuples should be of the form::
+
+                (offset, onoffseq),
 
-        %(Collection)s
+            where *onoffseq* is an even length tuple of on and off ink lengths
```

Comment:
I think odd-length sequences are allowed, see #17444. Or is that specifically not supported in collections?

Reply:
This matches what is currently said in `Line2D.set_dashes` and `Line2D.set_linestyle`. I see that we now accept odd-length sequences in our backends, but all the user-facing documentation says otherwise. Are we sure that there's not other third-party backends that assume even length? 

I'd personally prefer to make a separate PR fixing this since I'll want to fix it in 4-5 places throughout the documentation to more correctly describe how we do dashes, if that's okay.

Reply:
EDIT: 12+ places where this needs changes. Would definitely prefer to do this in a separate PR.